### DISCUSSION TITLE
how do chips work?

### DISCUSSION THREAD

**Discussion by aep** [Blackymas/NSPanel_HA_Blueprint#1005] [Q&A]
Heya,

thanks for this amazing thing.

i'm confused how chips work.
it looks like these only work with a specific entity type, but it's not clear which. 
the only thing i found that works is my ubiquiti wifi state.
but no switches work

possibly this needs some specific setup in HA?

i'm  sure this is not related to https://github.com/Blackymas/NSPanel_HA_Blueprint/discussions/557
since i have no errors in the log, and the one entity works.

**Comment by edwardtfn** [comment]
So, let's take this screenshot of home page as reference:

![](https://user-images.githubusercontent.com/41958506/212768891-9610d800-43c5-454f-a8b6-6b7c43677b5d.png)

Those small icons on the very top are what we call "Chips". They are 10 icons in total, but the first 3 are reserved for the relays and the climate entity, so the other 7 are the chips you can select from the Blueprint settings.
Once selected, the chip will be visible if the entity's state is in enable/on/true/etc, and won't be visible when those are disabled/off/false/etc.
The chips are visible only on Home page and there are no controls, which means clicking on a chip has no action.

The icons below the time and date are called "Home page buttons". Those are not selectable by users, and will be visible when notification, QR code, entities pages and/or alarm (coming soon) are enabled, otherwise they are not visible.

**Reply by ra-kal** [reply]
Could you also make it possible for numerical entities to show the icon, when value is non-zero (or a defined value).
I know, I can create a template for that, but ...
Usage: show icon, when power consumption > 0W (or defined value)

**Reply by grigi** [reply]
You could create a template sensor in HA that has that logic, and presents a True/False for that.
